lib/m2ts/caption-stream.js

```diff
@@ -406,10 +411,45 @@ Cea708Service.prototype.setCurrentWindow = function(windowNum) {
   this.currentWindow = this.windows[windowNum];
 };
 
-var Cea708Stream = function() {
+/**
+ * Try to create a TextDecoder if it is natively supported
+ */
+Cea708Service.prototype.createTextDecoder = function(encoding) {
+  if (typeof TextDecoder === 'undefined') {
+    this.trigger('log', {
+      level: 'warn',
+      message: 'The `encoding` option is unsupported without TextDecoder support'
+    });
+  } else {
+    try {
+      this.textDecoder_ = new TextDecoder(encoding);
+    } catch (error) {
+      this.trigger('log', {
+        level: 'warn',
+        message: 'TextDecoder could not be created for provided `encoding` value: ' + encoding
+      });
+    }
+  }
+}
+
+var Cea708Stream = function(options) {
```

Comment:
we should add a null check for options, should help with a lot of the test failures
```suggestion
var Cea708Stream = function(options) {
  var options = options || {};
```